lib/improver/cli/apply_emos_coefficients.py

```diff
@@ -0,0 +1,202 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to apply coefficients for Ensemble Model Output
+Statistics (EMOS), otherwise known as Non-homogeneous Gaussian
+Regression (NGR)."""
+
+import numpy as np
+
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.argparser import ArgParser
+from improver.ensemble_calibration.ensemble_calibration import (
+    ApplyCoefficientsFromEnsembleCalibration)
+from improver.ensemble_copula_coupling.ensemble_copula_coupling import (
+    EnsembleReordering,
+    GeneratePercentilesFromMeanAndVariance,
+    GeneratePercentilesFromProbabilities,
+    GenerateProbabilitiesFromMeanAndVariance,
+    RebadgePercentilesAsRealizations,
+    ResamplePercentiles)
+from improver.utilities.cube_checker import find_percentile_coordinate
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main(argv=None):
+    """Load in arguments for applying coefficients for Ensemble Model Output
+       Statistics (EMOS), otherwise known as Non-homogeneous Gaussian
+       Regression (NGR). The coefficients are applied to the forecast
+       that is supplied, so as to calibrate the forecast. The calibrated
+       forecast is written to a netCDF file.
+    """
+    parser = ArgParser(
+        description='Apply coefficients for Ensemble Model Output '
+                    'Statistics (EMOS), otherwise known as Non-homogeneous '
+                    'Gaussian Regression (NGR)')
+    # Filepaths for the forecast, EMOS coefficients and the output.
+    parser.add_argument('forecast_filepath', metavar='FORECAST_FILEPATH',
```

Comment:
Need to say this can be in percentiles, probabilities or realizations and that the output will be in the same form. In general the description could maybe have a bit more information about the steps the different input forms will go through. What do you think?

Reply:
I've added a bit more information here.